packages/expo-location/android/src/main/java/expo/modules/location/taskConsumers/GeofencingTaskConsumer.java

```diff
@@ -0,0 +1,252 @@
+package expo.modules.location.taskConsumers;
+
+import android.app.PendingIntent;
+import android.app.job.JobParameters;
+import android.app.job.JobService;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.PersistableBundle;
+import android.util.Log;
+
+import com.google.android.gms.location.Geofence;
+import com.google.android.gms.location.GeofenceStatusCodes;
+import com.google.android.gms.location.GeofencingClient;
+import com.google.android.gms.location.GeofencingEvent;
+import com.google.android.gms.location.GeofencingRequest;
+import com.google.android.gms.location.LocationServices;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import expo.interfaces.taskManager.TaskConsumer;
+import expo.interfaces.taskManager.TaskManagerUtilsInterface;
+import expo.interfaces.taskManager.TaskConsumerInterface;
+import expo.interfaces.taskManager.TaskInterface;
+import expo.modules.location.LocationHelpers;
+import expo.modules.location.LocationModule;
+
+public class GeofencingTaskConsumer extends TaskConsumer implements TaskConsumerInterface {
+  private static final String TAG = "GeofencingTaskConsumer";
+
+  private TaskInterface mTask;
+  private PendingIntent mPendingIntent;
+  private GeofencingClient mGeofencingClient;
+  private GeofencingRequest mGeofencingRequest;
+  private List<Geofence> mGeofencingList;
+  private Map<String, PersistableBundle> mRegions;
+
+  public GeofencingTaskConsumer(Context context, TaskManagerUtilsInterface taskManagerUtils) {
+    super(context, taskManagerUtils);
+  }
+
+  //region TaskConsumerInterface
+
+  public String taskType() {
+    return "geofencing";
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void didRegister(TaskInterface task) {
+    Context context = getContext();
+
+    if (context == null) {
+      Log.w(TAG, "The context has been abandoned.");
+      return;
+    }
+    if (!LocationHelpers.isAnyProviderAvailable(context)) {
+      Log.w(TAG, "There is no location provider available.");
+      return;
+    }
+
+    mTask = task;
+    mRegions = new HashMap<>();
+    mGeofencingList = new ArrayList<>();
+
+    // Create geofences from task options.
+    Map<String, Object> options = task.getOptions();
+    List<HashMap<String, Object>> regions = (ArrayList<HashMap<String, Object>>) options.get("regions");
+
+    for (Map<String, Object> region : regions) {
+      Geofence geofence = geofenceFromRegion(region);
+
+      if (geofence != null) {
```

Comment:
`geofenceFromRegion` will either throw an exception or return a `Geofence`, so this statement is always true and we should probably catch exception when calling `geofenceFromRegion`